src/services/db/RepoService.ts

```diff
@@ -242,6 +243,10 @@ export default class RepoService extends GitHubService {
     )
     const { private: isPrivate } = await super.getRepoInfo(sessionData)
 
+    if (targetFile === undefined) {
```

Comment:
perf: we can reduce a redundant `super.getRepoInfo(sessionData)` in this control flow by adding this line before `const { private: isPrivate } = await super.getRepoInfo(sessionData)`

Reply:
Oh good idea, updated in ad2521fd.